### DISCUSSION TITLE
GitHub Pages Builds Fail

### DISCUSSION THREAD

**Discussion by carterchannui** [community/community#22007] [Other Feature Feedback, Questions, & Ideas]
<p>I’m trying to create a website using Github Pages. The website has been building fine in the past. Today I updated some of the html and the website wouldn’t build anymore. I tried reverting to the previous commit, but it wouldn’t build either. I read the troubleshooting page, to no avail. I don’t have any idea of what to do anymore, please help.</p>
<p>The error:</p>
<pre><code>Your site is having problems building: Unable to build page. Please try again later.
</code></pre>
<p>My repository: <a href="https://github.com/carterchannui/integratedesports.com" rel="nofollow noopener">https://github.com/carterchannui/integratedesports.com</a> </p>

**Comment by SethClydesdale** [comment]
<p>Hello <a class="mention" href="/carterchannui">@carterchannui</a>,</p>
<p>If you’re not using <a href="https://jekyllrb.com/docs/github-pages/" rel="nofollow noopener">Jekyll</a> to build your website, sometimes GitHub pages will mistake certain characters, specifically curly braces, as jekyll objects, tags, etc. causing your page build to fail. If you’re just using static HTML pages (which is what it looks like to me), the easiest and quickest fix would be to disable Jekyll processing, which can be done by creating a file named <strong>.nojekyll</strong> in the root of your pages repository.</p>

**Comment by carterchannui** [comment]
<p>Hi <a class="mention" href="/sethclydesdale">@sethclydesdale</a>, </p>
<p>If by static you mean without animations, I don’t believe my website is just static HTML. This is what the website looks like: <a href="https://integratedesports.com/#hero" rel="nofollow noopener">https://integratedesports.com/#hero</a>. Also, why has it just started failing? It wasn’t doing this before. </p>

**Comment by SethClydesdale** [comment, marked as answer]
<p>By static, I meant a website that’s created with HTML (no back-end coding). <a href="https://techterms.com/definition/staticwebsite" rel="nofollow noopener">This</a> has a pretty good explanation. Anyway, checking your repo, I’m seeing errors like the one below.</p>
<pre><code>The variable `{{a}` on line 50 in `materialize-admin-template/node_modules/balanced-match/README.md` was not properly closed with `}}`.
</code></pre>
<p>I’m assuming it may be mistaken for Jekyll liquid code. If you’re not using Jekyll, I would suggest creating a file named  <strong>.nojekyll</strong> (it doesn’t need any contents) in the root of your repository to see if it resolves the errors. If you are using Jekyll, then handling the file in the error above should resolve the issue as well.</p>

**Comment by carterchannui** [comment]
<p>Thank you <a class="mention" href="/sethclydesdale">@sethclydesdale</a>, creating the  <strong>.nojekyll</strong> file effectively worked. The website has updated and was built successfully. GitHub did send me an email that I would like understand:</p>
<pre><code>The page build completed successfully, but returned the following warning for the `master` branch:

It looks like you're using GitHub Pages to distribute binary files. We strongly suggest that you use releases to ship projects on GitHub. Releases are GitHub's way of packaging and providing software to your users. You can think of it as a replacement to using downloads to provide software. We found the following file(s) which may be a good candidate for releases: materialize-admin-template/node_modules/gifsicle/vendor/gifsicle.exe, materialize-admin-template/node_modules/jpegtran-bin/vendor/jpegtran.exe, materialize-admin-template/node_modules/node-notifier/vendor/notifu/notifu.exe, materialize-admin-template/node_modules/node-notifier/vendor/notifu/notifu64.exe, materialize-admin-template/node_modules/node-notifier/vendor/snoreToast/SnoreToast.exe, and materialize-admin-template/node_modules/optipng-bin/vendor/optipng.exe. For more information, see https://help.github.com/en/articles/about-releases.

For information on troubleshooting Jekyll see:

  https://help.github.com/articles/troubleshooting-jekyll-builds

If you have any questions you can contact us by replying to this email.
</code></pre>
<p>What does this mean? Is it telling me to switch to a different program? </p>

**Comment by SethClydesdale** [comment]
<p>Good to hear!</p>
<p>No, it’s more of a suggestion to keep large binary files (.exe, for example) to the releases section, since they can go towards the recommended <a href="https://help.github.com/en/articles/what-is-my-disk-quota" rel="nofollow noopener">1GB limit for repositories</a>. If the files are necessary for your website to function then you can keep them as is. If not, then you can always remove them to keep your repo’s size down.</p>

**Comment by wesleytodd** [comment]
<p>The documentation of this feature is a bit missleading.  Is the <code>.nojekyll</code> file supposed to be in the root of the <code>gh-pages</code> branch, or in the root of the repository <code>master</code> branch?</p>
<p>If in the master branch, it seems to me that it should also be possiable to put it in the root of whereever the pages is served from.  But that is just my opinion :slight_smile:</p>

**Comment by hadeelkh** [comment]
<p>I had that problim too, and this solution solved it … thanks alot</p>

**Comment by KingsleyMcSimon** [comment]
<p>I had the same problem and was glad that the suggestion creating a .nojekyll file solved the problem and my site was successfully published.</p>

**Comment by austin-sanga** [comment]
<p>I am having a problem building</p>

**Comment by DGCK81LNN** [comment]
<p>I’m having the same problem and I already have a <code>.nojekyll</code> in my repo. It seems this is a different problem. See this topic: <a href="https://github.community/t/my-github-pages-stopped-to-build/180173/2">https://github.community/t/my-github-pages-stopped-to-build/180173/2</a></p>

**Comment by Firoz01** [comment]
<p>thank you… I had the same problem… But I fixed it by saw your solution. many many thanks</p>

**Comment by tech-teni** [comment]
<p>this actually solved my problem, Thanks</p>
